docs/specification.md

```diff
@@ -182,9 +182,9 @@ Usually, a Tag represents a concept of the domain, e.g. the type and id of an en
 The Append Condition is used to enforce consistency, ensuring that between the time of building the Decision Model and appending the events, no new events were stored by another client that match the same query.
 
 - It _MUST_ contain a `failIfEventsMatch` [Query](#query)
-  - this is typically the same Query that was used when building the Decision Model
 - It _MAY_ contain an `after` [Sequence Position](#sequence-position)
   - this represents the highest position the client was aware of while building the Decision Model. The Event Store _MUST_ ignore the Events before the specified position while checking the condition for appending events. _Note:_ This number can be _higher_ than the position of the last event matching the Query.
+  - when present, the `failIfEventsMatch` Query is typically the same Query that was used when building the Decision Model, which guarantees that the Decision Model is still the same when we append new events
```

Comment:
I really like the "which guarantees that the Decision Model is still the same when we append new events" part, that makes it clearer.
But the "when present" part is not entirely correct:
The signature of the `append` endpoint is basically
```ts
append(events: Events|Event, condition?: { failIfEventsMatch: Query, after?: SequencePosition })
```

So the optional parts are `condition` and `condition.after`. i.e. `failIfEventsMatch` can't be omitted

Reply:
Ha, it's a bullet under "It my contain `after`, I meant "when `after` is present", implicitly. It better be explicit.